locksmith/src/controllers/v2/grantKeysController.ts

```diff
@@ -0,0 +1,62 @@
+import { Request, Response } from 'express'
+import Normalizer from '../../utils/normalizer'
+import logger from '../../logger'
+import Dispatcher from '../../fulfillment/dispatcher'
+
+export class GrantKeysController {
+  constructor() {}
+
+  /**
+   * API to grant keys, can only be called by a lock manager with an "app token".
+   * URL includes:
+   * - network
+   * - lock
+   * Body must include list of keys to be granted in the form
+   * [
+   *    {
+   *      recipient
+   *      expiration
+   *      manager
+   *    }
+   * ]
+   * @param request
+   * @param response
+   * @returns
+   */
+  async grantKeys(request: Request, response: Response) {
+    try {
+      const lockAddress = Normalizer.ethereumAddress(request.params.lockAddress)
+      const network = Number(request.params.network)
+      const keys = request.body.keys
+
+      /** Duration and managers are ignored for now, until a later PR since dispatcher does not support them yet */
+      const recipients = keys.map((k: any) => k.recipient)
+      const dispatcher = new Dispatcher()
+
+      dispatcher.grantKeys(
+        lockAddress,
+        recipients,
+        network,
+        async (error: any, hash: string) => {
+          if (error) {
+            response.status(500).send({
+              error,
+            })
```

Comment:
I would also attach a message field with a useful but generalised error here. It can be hard to know what error spits out.

Reply:
Done :)